### ISSUE TITLE
Drawers, Drawer Controller and Magical Crop Seeds

### ISSUE THREAD

**Issue by Redhandjane** [jaquadro/StorageDrawers#97]
I'm running a server and we're trying to use drawers and the controller for our sorting house. When we are using it in the garden area, we have drawers that have seeds from Magical Crops, namely gold, nether, ender, diamond and seemly the other high end seeds that won't go into the drawer. We're piping them in with Ender IO's item conduit to the drawer controller. This works with vanilla seeds and most of the lower tier magical crop seeds. But I have stacks of the higher tier seeds that won't go into their drawer, either by controller, conduit, or manually. I'm not getting any kind of error. Please let me know if there is something somewhere I can get for you to help you with this issue.

**Comment by jaquadro** [comment]
Are these going into compacting drawers or normal drawers?  Are there any expected ore dictionary conversions?

**Comment by Redhandjane** [comment]
Normal drawers and I'm not sure what you mean by expected ore dictionary conversions?

**Comment by jaquadro** [comment]
If you have 4 kinds of iron nugget for example, the drawer would convert them for you to the one already stored, unless there was something strange with your world's ore dictionary.

But you mentioned seeds.  Are you using AgriCraft?  If you pull some seeds out of the drawer, can you stack them with the ones you're trying to insert?

**Comment by Redhandjane** [comment]
No, we're not using AgriCraft, the only farming mod we have is Magical Crops. The drawers are doing fine with the nuggets/ingots, but it's the seeds that's not working. When I pull the seeds out, they do stack but when I try to put them back in, it'll only take the first stack or two then stop. 

The drawers are locked to that seed type and upgraded with the status indicators and the emerald size increase.

**Comment by jaquadro** [comment]
I'm not really familiar with magical crops, but are those high-value seeds supposed to stack to 64?

**Comment by Redhandjane** [comment]
Yes, and they will when I pull them out. Now I have the same issue with the rest of the seeds. I've restarted my server, but it hasn't helped.

**Comment by jaquadro** [comment]
If you create a new drawer, do you have any problems with the seeds?  Is it perhaps somethign that happens after you've been using EIO with them?

**Comment by Redhandjane** [comment]
Hmm. I think you're on to something. With new drawers, I can put all of the seeds on hand into it. If I take seeds from the area I have set up with EIO and try to put them into the new drawers, they won't go with the current stacks. I'll try using a different piping and let you know

**Comment by Redhandjane** [comment]
Yup, I think that's the problem. When they go through EIO something's happening and the drawers are recognizing the seeds properly. Seeds that are grown from new seeds aren't being sorted while seeds that have gone through EIO are sorting properly. 

I'm assuming since this is an EIO issue, I should let them know?

**Comment by jaquadro** [comment]
I'll take a look this evening.  It's premature to say that it's definitely EIO's problem, but it could be.  It sounds like some of the data on the ItemStack gets modified (or added to), but my hang-up with that is you can merge the stacks in your own inventory.

**Comment by jaquadro** [comment]
I tried doing a basic reproduction, but didn't find anything wrong.  Piping diamond seeds from a chest through EIO item conduit into a controller into a locked/upgraded drawer.  Didn't run into a problem where EIO stopped piping or I couldn't insert directly.

I think it will be difficult for either me or the EIO author to track this down (never mind he has a backlog of 400+ issues in his repo).  Can you reproduce this with just Storage Drawers, Ender IO, and Magical Crops installed?  

If you can produce a simple build that always reproduces the issue, a screenshot of that would be good.  Otherwise, I may need you to send me a copy of your world (pref. with most of the regions deleted) and mod set.

**Comment by Redhandjane** [comment]
Try setting up a system and piping in harvested seeds into drawers that have a couple spawned seeds in them.

I'll try reproducing this issue with just the 3 mods and let you know by tomorrow afternoon. Here are some screenshot, but I'm not sure how to send you a copy of the world (sorry, I can barely install mods lol). 
![2015-06-10_03 00 38](https://cloud.githubusercontent.com/assets/12817536/8076717/8c86a41c-0f1d-11e5-8669-44dc6335ac16.png)
![2015-06-10_03 00 53](https://cloud.githubusercontent.com/assets/12817536/8076719/90dc564c-0f1d-11e5-90c3-cca507b7b233.png)

**Comment by Redhandjane** [comment]
I've tried to recreate the issue in singleplayer with just Storage Drawers, EIO, and Magical Crops, however I have been unable to. I've also tried to use the full set of mods and still haven't been able to get it. I'm wondering if it could have been something with the server, our provider has taken the node down for critical repairs.

**Comment by jaquadro** [comment]
Really difficult to try and guess.

My advice at this point is if you still counter it somewhere in your world, you will need to send me a copy of that world.  Which is just zipping the world directory on the server and putting it up on a file share.  I would also want a zip of all your mods.

**Comment by Redhandjane** [comment]
Ok, I'll get you a copy of both as soon as i can access the server panel again.

**Comment by Redhandjane** [comment]
https://www.mediafire.com/?d10vl8mnbllhjkz

https://www.mediafire.com/?d10vl8mnbllhjkz

The first one is the map and the second is the mod. The chest to the left is full of seeds that won't go into the drawers. My whole sorting system just decided to randomly stop working. The only way I've found to fix it is to unlock the drawer and put the seeds from the chest into it and relock it.

**Comment by jaquadro** [comment]
Your second link appears to be the same as the first.

You mentioned unlocking: Are you saying that if you unlock the drawer, the same items that can't be manually inserted will then insert, without needing to first empty the drawer?

**Comment by Redhandjane** [comment]
Hah, so it is. 
http://www.mediafire.com/download/m8hs7tb38hf2ftw/Mods.zip

I have a wall of drawers set up that are locked to the items. If I unlock it, empty it, then put one of the seeds back in and relock it, it starts working again

**Comment by jaquadro** [comment]
I need the coordinates of the drawers.  The map placed me at spawn.  I found someone's base, and it has drawers, but it doesn't look like the one you were talking about.

**Comment by Redhandjane** [comment]
-411, 68, 417

That'll take you right to the sorting room. I didn't even think about adding the coords :/

**Comment by jaquadro** [comment]
So, I don't think the drawers are doing anything inherently wrong.

I saw that the seeds in the chest all had nonzero metadata values, but seeds that come from the creative menu do not.  And I can't drag and drop the different seeds on top of each other, but they do unify if I shift-click one of them within the inventory which is a bit strange.

In your farming area below, some of the chests have seeds hanging around, which also have metadata values.  That makes me think it's not related to EIO, and is either a specific behavior of magic crops (more likely), or related to golem harvesting (less likely).

I think you should take this up with the magic seeds dev.  Find out if this is an expected behavior (and if so, what the metadata means).  If this is normal and the seeds are expected to unify, then I will need to go back and do some special exceptions for the mod.  Under normal circumstances, merging items of different metadata values is a big no-no.

**Comment by Redhandjane** [comment]
Ok, I'll pass this message along. Thank you for looking into it!

**Comment by esotericist** [comment]
This is a longstanding issue I've found with magical crops and other odd seed mods. What SEEMS to be happening is when the crop is broken, it is failing to reset the metadata properly.

Most people don't notice because they get corrected in a few ways after the fact (most often automatic stack combining), but it's pretty easy to reproduce by just punching the crops at different stages of growth.
